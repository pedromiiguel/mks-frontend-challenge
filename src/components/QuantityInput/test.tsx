import { customRender } from '@/utils/tests/customRender';
import { screen, fireEvent } from '@testing-library/react';
import {
  decreaseItemQuantity,
  increaseItemQuantity
} from '@/features/cart/cart-slice';

import { QuantityInput } from '.';

describe('<QuantityInput />', () => {
  it('should render the quantity Input', () => {
    const props = { quantity: 1, productId: 1 };

    const { container } = customRender(<QuantityInput {...props} />);
    const increaseButton = screen.getByRole('button', {
      name: 'Aumentar a quantidade do produto'
    });
    const decreaseButton = screen.getByRole('button', {
      name: 'Diminuir a quantidade do produto'
    });

    expect(screen.getByText('Qtd:')).toBeInTheDocument();

    expect(increaseButton).toBeInTheDocument();
    expect(decreaseButton).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
  it('should call increaseItemQuantity when + button is clicked', () => {
    const productId = 1;

    const { store } = customRender(
      <QuantityInput quantity={3} productId={productId} />
    );
    store.dispatch = jest.fn();
    store.dispatch(increaseItemQuantity(productId));

    const increaseButton = screen.getByRole('button', {
      name: 'Aumentar a quantidade do produto'
    });

    fireEvent.click(increaseButton);
    expect(store.dispatch).toHaveBeenCalledWith(
      increaseItemQuantity(productId)
    );
  });

  it('should call decreaseItemQuantity when - button is clicked', () => {
    const productId = 2;

    const { store } = customRender(
      <QuantityInput quantity={2} productId={productId} />
    );
    store.dispatch = jest.fn();
    store.dispatch(decreaseItemQuantity(productId));

    const decreaseButton = screen.getByRole('button', {
      name: 'Diminuir a quantidade do produto'
    });

    fireEvent.click(decreaseButton);
    expect(store.dispatch).toHaveBeenCalledWith(
      decreaseItemQuantity(productId)
    );
  });
});
