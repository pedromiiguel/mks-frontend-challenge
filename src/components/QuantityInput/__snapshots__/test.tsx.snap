// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuantityInput /> should render the quantity Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: calc(0.8rem / 2);
}

.c1 {
  font-size: 1.2rem;
  color: #000000;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: calc(0.8rem / 2);
  border: 0.03rem solid #bfbfbf;
  border-radius: 0.5rem;
}

.c3 {
  border: 0;
  cursor: pointer;
  background-color: transparent;
  padding: calc(0.8rem / 2);
  line-height: 1;
  font-size: 1.6rem;
  font-weight: 400;
  color: #000000;
}

.c4 {
  border: 0;
  width: 2.6rem;
  padding-inline: calc(0.8rem / 2);
  border-left: 0.02rem solid #bfbfbf;
  border-right: 0.02rem solid #bfbfbf;
  color: #000000;
}

@media (max-width:450px) {
  .c1 {
    display: none;
  }
}

@media (max-width:450px) {
  .c3 {
    font-size: 2.0rem;
    padding: calc(1.2rem / 2);
  }
}

@media (max-width:450px) {
  .c4 {
    width: 3.0rem;
    padding-inline: 0.8rem;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
    for="quantity"
  >
    Qtd:
  </label>
  <div
    class="c2"
  >
    <button
      aria-label="Aumentar a quantidade do produto"
      class="c3"
    >
      -
    </button>
    <input
      class="c4"
      id="quantity"
      readonly=""
      type="text"
      value="1"
    />
    <button
      aria-label="Diminuir a quantidade do produto"
      class="c3"
    >
      +
    </button>
  </div>
</div>
`;
